src/Compilers/CSharp/Test/Semantic/Semantics/RefExtensionMethodsTests.cs

```diff
@@ -0,0 +1,438 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
+{
+    public class RefExtensionMethodsTests : CSharpTestBase
+    {
+        [Fact]
+        public void ExtensionMethods_RValues_Ref()
+        {
+            var code = @"
+public static class Extensions
+{
+    public static void PrintValue(ref this int p)
```

Comment:
Add a test when both `PrintValue(ref ...) `  and `PrintValue(in ...)` are available - coming from different extension classes. We should not be able to call either, regardless is `this` is an RValue or LValue.
NOTE: when calling directly, as a regular method, we can disambiguate by passing the arg as `ref`, but in extension method case we can't.